src/shogun/base/SGObject.h

```diff
@@ -39,6 +39,30 @@ class Parameter;
 class CSerializableFile;
 class ParameterObserverInterface;
 
+// all shogun base classes for put/add templates
```

Comment:
great but i recon since here is everything a fwd declaration as SGObject.h is already quite cramped with stuff maybe it'd be better to factor this out somewhere else and just include that header in this header